tests/test_s3.py

```diff
@@ -0,0 +1,15 @@
+from patronload.s3 import delete_objects_from_bucket_with_prefix
+
+
+def test_delete_objects_from_bucket_with_prefix_deletes_only_expected_objects(
+    caplog, mocked_s3, s3_client
+):
+    assert len(mocked_s3.list_objects(Bucket="test-bucket")["Contents"]) == 2
```

Comment:
`list_objects_v2` here. Also I would confirm that this count is 1 after the deletion.